ARrehab/ARrehab/Tile.swift

```diff
@@ -12,19 +12,90 @@ import CoreGraphics
     
+    //Class attributes
+    static let TILE_SIZE = SIMD3<Float>(0.5, 0.01, 0.5)
+    static let TILE_COLLISION_GROUP = CollisionGroup(rawValue: 1)
+   
+    static let defaultTileModel = ModelComponent(mesh: MeshResource.generateBox(size: Tile.TILE_SIZE, cornerRadius: 0.2), materials: [SimpleMaterial()])
+    static let defaultCollisionComp = CollisionComponent(shapes: [ShapeResource.generateBox(width: 0.5, height: 4.0, depth: 0.5)], mode: .trigger, filter: .sensor)
+    
+    //Instance variables
     var tileName: String
+    let coords : Coordinates
+    
     
+    //Default initializer: Uses default model and collision componenets, does not adjust translation
     required init(name: String, x: Float, z: Float) {
         self.tileName = name
+        self.coords = Coordinates(x: x, z: z)
+        
         super.init()
-        self.components[ModelComponent] = ModelComponent(mesh: MeshResource.generateBox(width: 0.5, height: 0.01, depth: 0.5, cornerRadius: 0.2), materials: [SimpleMaterial()])
-        self.components[CollisionComponent] = CollisionComponent(shapes: [ShapeResource.generateBox(width: 0.5, height: 4.0, depth: 0.5)], mode: .trigger, filter: .sensor)
-        self.transform.translation = SIMD3<Float>(x,0.0,z)
+
+        self.components[ModelComponent] = Tile.defaultTileModel
+        self.components[CollisionComponent] = Tile.defaultCollisionComp
+        self.collision?.filter.group = Tile.TILE_COLLISION_GROUP
+        
+        self.transform.translation = SIMD3(x, 0.0, z)
         print("Generated Tile: " + name)
+    }
+    
+    //Optional initializer: Allows specification of custom model components and translation adjustment
+    convenience init(name: String, x: Float, z: Float, materials: [Material]?, adjustTranslation: Bool = false) {
+        /*self.tileName = name
+        
+        self.coords = Coordinates(x: x, z: z)
+        
+        super.init()
+
+        self.components[ModelComponent] = modelComp
+        self.components[CollisionComponent] = Tile.defaultCollisionComp
+        self.collision?.filter.group = Tile.TILE_COLLISION_GROUP*/
+        
+        self.init(name: name, x: x, z: z)
+        
+        if materials != nil {
+            self.changeMaterials(materials: materials!)
+        }
+        
+        if adjustTranslation {
+            self.transform.translation = Tile.adjustTransformTranslation(coords: self.coords)
+        }
         
+        print("Generated Tile: " + name)
     }
     
     required init() {
-        fatalError("init() has not been implemented")
+        fatalError("Can't instantiate a Tile with no paramaters")
+    }
+}
+
+
+//Nested data types and convenience methods
+extension Tile {
+    
+    struct Coordinates : Hashable, Equatable {
+        var x : Float
+        var z : Float
+    }
+    
+    //Adjusts the translation of the tiles so that the appropriate vertex of the tile is at the coordinates it is initialized with (rather than the center)
```

Comment:
are the coordinates of the Tiles placed relative to the center of the board?

From what I understand, this function:
/** Adjusts the translation of the tiles such that the specified coordinates are the coordinates of the corner farthest from the center of the board.
- Parameters:
    - coords: coordinates of the corner of the tile farthest from the center of the board.
*/

Also why are tiles with coordinates (0,0) assumed to be adjusted such that its in the lower left hand side?

Note I should probably read on to find out more...

Reply:
No, coordinates of the tiles are relative to the center of the plane (grid generation starts in the upper right corner of the surface plane and works down/across from there). So the center of the eventual grid/board is not necessarily the center of the plane, and there is no real reference to it. I assume that answers your question about adjusting (0,0) as well?

Reply:
Hm I'm still not very clear on how you choose which vertex to be the "appropriate" vertex.

Reply:
In the interests of time I'm just gonna resolve everything and assume that prior testing has ensured that it works. We can go over it again in the future if needed.